### PR TITLE
Implemented get_table_by_scope

### PR DESCRIPTION
**Pull request by aaroncox**

### CODE REVIEW

File: src/api/v1/types.ts [hunk 1]
```diff
@@ -370,3 +370,28 @@ export interface GetTableRowsResponse<Index = TableIndexType, Row = any> {
     ram_payers?: Name[]
     next_key?: Index
 }
+
+export interface GetTableByScopeParams<Index = TableIndexType | string> {
```
Comment (jnordberg) [inline src/api/v1/types.ts:374]: There is no index type or payer:
```
struct get_table_by_scope_params {
      name        code; // mandatory
      name        table; // optional, act as filter
      string      lower_bound; // lower bound of scope, optional
      string      upper_bound; // upper bound of scope, optional
      uint32_t    limit = 10;
      optional<bool>  reverse;
   };
```

File: src/api/v1/types.ts [hunk 2]
```diff
@@ -370,3 +370,28 @@ export interface GetTableRowsResponse<Index = TableIndexType, Row = any> {
...
+    @Struct.field('name') scope!: Name
+    @Struct.field('name') table!: Name
+    @Struct.field('name') payer!: Name
+    @Struct.field('uint32') count!: UInt32
+}
+
+@Struct.type('get_table_by_scope_response')
+export class GetTableByScopeResponse extends Struct {
+    @Struct.field(GetTableByScopeResponseRow, {array: true}) rows!: GetTableByScopeResponseRow[]
+    @Struct.field('name') more!: Name
```
Comment (jnordberg) [inline src/api/v1/types.ts:396]: `more` is a string

```
   struct get_table_by_scope_result {
      vector<get_table_by_scope_result_row> rows;
      string      more; ///< fill lower_bound with this value to fetch more rows
   };
```

**Review by jnordberg — CHANGES_REQUESTED** [review]

File: src/api/v1/types.ts (hunk 1 above)
Comment (aaroncox) [reply src/api/v1/types.ts:374]: lol, then I shouldn't be following the official documentation:

https://developers.eos.io/manuals/eos/latest/nodeos/plugins/chain_api_plugin/api-reference/index#operation/get_table_by_scope

File: src/api/v1/types.ts (hunk 2 above)
Comment (aaroncox) [reply src/api/v1/types.ts:396]: It is returning a name value, as in an account. Shouldn't we type that as a name?

**Comment by jdheeter** [comment]
Any reason not to merge this?

**Comment by aaroncox** [comment]
I had the changes Johan requested, but they weren't pushed up until now.

**Review by jnordberg — APPROVED** [review]
👍

**Comment by jnordberg** [comment]
Published as `0.4.3`

**Pull request merged**